src/components/OTP.vue

```diff
@@ -0,0 +1,337 @@
+     * - if 'Request OTP' button is enabled
+     * - if OTP hasn't been typed yet
+     */
+    isSubmitButtonDisabled() {
+      return !(this.isRequestOTPButtonDisabled || this.OTPCode.length > 0);
+    },
+
+    /** Checks if the current input entry has the required number of characters */
+    isCurrentEntryIncomplete() {
+      return this.phoneNumber.length < this.maxLengthOfId;
+    },
+
+    /** Whether input being typed is in the correct format */
+    isInvalidInputMessageShown() {
+      return this.invalidInputMessage != null;
+    },
+
+    /** Returns the heading text for the input box */
+    inputBoxDisplayText() {
+      return this.programData.text.default.display;
+    },
+
+    /** Returns the text for the OTP code input box */
+    enterOTPInputBoxDisplayText() {
+      return this.programData.text.default.enterOTP;
+    },
+
+    /** Returns the text for the button to request an OTP */
+    requestOTPButtonDisplayText() {
+      return this.programData.text.default.requestOTP;
+    },
+
+    /** Returns the text for the submit button */
+    submitButtonDisplayText() {
+      return this.programData.text.default.submitButton;
+    },
+
+    /** Returns the maximum length of the ID */
+    maxLengthOfId() {
+      return this.programData.input.maxLengthOfId;
+    },
+
+    /** Returns the basic validation type for the input */
+    basicValidationType() {
+      return this.programData.input.basicValidationType;
+    },
+
+    /** Returns the invalid input message stored against each program */
+    invalidInputText() {
+      return this.programData.text.default.invalid.input;
+    },
+  },
+  methods: {
+    /** Determines how the input box should look.
+     * - If an input error needs to be displayed, the box has a a red border.
+     * - Otherwise, it has an opacity of 30.
+     */
+    selectInputBoxClasses() {
+      let baseStyle =
+        "border-2 rounded-sm p-4 mx-auto border-gray-500 focus:border-gray-800 focus:outline-none";
+      return this.invalidInputMessage
+        ? baseStyle + "border-red-600 focus:border-red-600"
+        : baseStyle + "pointer-events-none opacity-30";
+    },
+
+    /** Calls the mapping function to validate the typed character
+     * @param {Object} event - event triggered when a character is typed
+     */
+    isValidPhoneNumber(event) {
+      if (validationTypeToFunctionMap[this.basicValidationType](event)) {
+        return true;
+      } else event.preventDefault();
+    },
+
+    /** Resets the invalid input message */
+    resetInvalidInputMessage() {
+      this.invalidInputMessage = "";
+    },
+
+    /** This function is called whenever something is entered in the input box.
+     * It checks if the required number of characters are being typed.
+     * @param {Object} event - the event which triggered this function
+     */
+    updatephoneNumber(event) {
+      if (event.target.value.length == 0) {
+        this.invalidInputMessage = "";
```

Comment:
`inputMessage` refers to what here? I think since here we are dealing with only phone numbers, a lot of the variable names can be simplified, like @deepansh96 suggested

Reply:
It's the message that keeps being displayed until the user types the required number of characters. "Please type 10 digits."

Reply:
Point here was that variable names can be simplified, since this is the OTP component and will clearly deal only with phone numebrs and OTPs